pyp2rpm/convertor.py

```diff
@@ -157,28 +157,26 @@ def metadata_extractor(self):
             raise AttributeError(
                 'local_file attribute must be set before calling metadata_extractor')
 
-        if not hasattr(self, '_metadata_extractor'):
-            if self.pypi:
-                logger.info('Getting metadata from PyPI.')
-                self._metadata_extractor = metadata_extractors.PypiMetadataExtractor(
-                    self.local_file,
-                    self.name,
-                    self.name_convertor,
-                    self.version,
-                    self.client,
-                    self.rpm_name,
-                    self.venv,
-                    self.base_python_version)
-            else:
-                logger.info('Getting metadata from local file.')
-                self._metadata_extractor = metadata_extractors.LocalMetadataExtractor(
-                    self.local_file,
-                    self.name,
-                    self.name_convertor,
-                    self.version,
-                    self.rpm_name,
-                    self.venv,
-                    self.base_python_version)
+        if self.local_file.endswith('.whl'):
+            logger.info('Getting metadata from wheel using WheelMetadataExtractor.')
+            self._metadata_extractor = metadata_extractors.WheelMetadataExtractor(
+                self.local_file,
+                self.name,
+                self.name_convertor,
+                self.version,
+                self.rpm_name,
+                self.venv,
+               self.base_python_version)
+        else:
+            logger.info('Getting metadata from setup.py using DistMetadataExtractor.')
+            self._metadata_extractor = metadata_extractors.DistMetadataExtractor(
+                self.local_file,
+                self.name,
+                self.name_convertor,
+                self.version,
+                self.rpm_name,
+                self.venv,
+               self.base_python_version)
```

Comment:
Wouldn't it be cleaner to save the class to variable a and call the variable? Something like:

    if ...:
        extractor_cls = metadata_extractors.WheelMetadataExtractor
    else:
        extractor_cls = metadata_extractors.DistMetadataExtractor
    self._metadata_extractor = extractor_cls(
        self.local_file,
        self.name,
        self.name_convertor,
        self.version,
        self.rpm_name,
        self.venv,
        self.base_python_version)

There is also inconsistent usage of spaces/indent (lines 169 and 179).